docs/decisions/0002-authorization-model-foundation.rst

```diff
@@ -0,0 +1,195 @@
+0002: Authorization (AuthZ) Model Foundations
+#############################################
+
+Status
+******
+**Draft**
+
+Context
+*******
+Open edX needs a single way to decide: who can do what, on which resource, and under which conditions. Today, permissions are checked in many different ways. Some systems are feature-specific (``student_courseaccessrole``, ``django_comment_client_role``, ``contentlibrarypermission``). Others use global roles passed in JWTs. Many checks are written directly in code (``if user.is_superuser``). This makes the system hard to extend, hard to change, and not easy to audit.
+
+We want an authorization model that is:
+
+* Clear and consistent vocabulary everywhere.
+* Explicitly supports industry standards and is built on battle-tested technologies.
+* Flexible but still simple to maintain.
+* Able to explain every decision (the system should be transparent on why access was granted or not).
+* Unified and centralized enforcement rather than ad-hoc implementations for immediate needs.
+* Able to support query-based access patterns out of the box.
+* Focused on connecting stakeholders and making policies clear and accessible to everyone involved.
+
+.. note::
+
+   Authorization is considered independent from authentication. There will be an interface between them so we can combine correctness and consistency. A separate ADR will cover the details of this interface (e.g., how roles in JWTs are handled and how checks are made).
+
+Decision
+********
+
+I. Canonical Permission Model
+=============================
+
+Normalize all checks to Subject-Action-Object-Context (S-A-O-C)
+----------------------------------------------------------------
+* We express authorization as: is **Subject** allowed to do **Action** on **Object** under **Context**?
+* This normalization is used in policies, code, queries, and audits.
+* Examples:
+
+  - Can Alice (subject) edit (action) Course 123 (object) as part of Org A (context)?
+  - Can a service account (subject) read (action) Library X (object) during maintenance mode (context)?
+
+II. Resources and Scopes
+========================
+
+Scopes as first-class citizens in permission-granting
+-----------------------------------------------------
+* A **scope** defines the boundary within which a role or policy applies (for example: platform-wide, organization-wide, a single course, or a specific library).
+* Treating scopes as **first-class citizens** means they are explicitly modeled in the system, not hidden inside ad-hoc resource definitions. They must be available to policies, queries, and audits in a consistent way.
+* Scopes can be **parameterized** (e.g., ``organization:ORG-A``, ``course:CS101``,  ``site:sandbox.openedx.org``, ``instance``) to support granular checks.
```

Comment:
[trying to understand] Is `site` included to support needs of multitenancy?

Reply:
The question of whether we're going to support, eventually, or at least leave the door open for others to support it still stands cause I think the model is flexible enough to do so. We can go with official support for: instance-wide, orgs, courses, course-runs, and libraries for the now but I don't think we're limited to this scopes.